replace2.js

```diff
@@ -729,92 +782,84 @@ transform.render = async (obj, divId, previousResults = {}) => {
   contents = contents.replace(//g, "");
   // add the HTML/HEAD/BODY tags...
   document.getElementById(divId).innerHTML =
-    contents +
-    `
-  <div class="modal" id="softModal" tabindex="-1" role="dialog">
+    contents + `
+    <div class="modal" id="softModal" tabindex="-1" role="dialog" aria-labelledby="softModalTitle" aria-modal="true">
       <div class="modal-dialog" role="document">
-          <div class="modal-content">
+        <div class="modal-content">
           <div class="modal-header">
-              <h5 class="modal-title">Response Requested</h5>
-              <button type="button" class="close" data-dismiss="modal" data-bs-dismiss="modal" aria-label="Close">
-              <span aria-hidden="true">&times;</span>
-              </button>
+            <h5 class="modal-title" id="softModalTitle" tabindex="-1">Response Requested</h5>
+            <button type="button" class="close" data-dismiss="modal" data-bs-dismiss="modal" aria-label="Close">
+              <span aria-hidden="true">×</span>
+            </button>
           </div>
-          <div id="modalBody" class="modal-body">
-              <p id="modalBodyText">There is 1 unanswered question on this page. Would you like to continue?</p>
+          <div id="modalBody" class="modal-body" aria-describedby="modalBodyText">
+            <p id="modalBodyText">There is 1 unanswered question on this page. Would you like to continue?</p>
           </div>
           <div id="softModalFooter" class="modal-footer">
-              <button type="button" id=modalContinueButton class="btn btn-light" data-dismiss="modal" data-bs-dismiss="modal">Continue Without Answering</button>
-              <button type="button" id=modalCloseButton class="btn btn-light" data-dismiss="modal" data-bs-dismiss="modal">Answer the Question</button>
-          </div>
+            <button type="button" id="modalContinueButton" class="btn btn-light" data-dismiss="modal" data-bs-dismiss="modal">Continue Without Answering</button>
+            <button type="button" id="modalCloseButton" class="btn btn-light" data-dismiss="modal" data-bs-dismiss="modal">Answer the Question</button>
           </div>
+        </div>
       </div>
-  </div>
-  <div class="modal" id="hardModal" tabindex="-1" role="dialog">
+    </div>
```

Comment:
maybe temporarily sideline these changes, I'm moving these modals out of here an importing them in to clean up the code

Reply:
Okay, these changes make a big difference in navigating that modal with a screen reader (the loading/nav/return sequence was pretty difficult without them).

Do you want to remove this code once yours are in place, do you want me to revert what I added, or do you want it removed entirely?

Reply:
All of the hardcoded modals (four in total).

Would it be too much to pre-maturely implement where I have these modals moved and throw all your changes on top of that? If so, put your stuff in and I can rework my changes when it's time for me to push. If not, you can find what I've done on the `spanish-implementation` branch. The modals have all been moved to a new file called `common.js`.

Reply:
Just to follow up, I haven't had a chance to look at this yet. I ran into some prod Home Collections issues today and a loading issue with a SOCcer question in local testing—761310265 (Module 1)—response options are not loading. I will update you as soon as possible, but just FYI, I haven't familiarized myself with what you're requesting here yet.

Reply:
Let's connect tomorrow about the SOCcer issue if you're still having problems.

Reply:
I left these as-is based on our Teams convo. Thanks